bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/scanner/RowMergerTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved. Licensed under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with the License. You may obtain
+ * a copy of the License at http://www.apache.org/licenses/LICENSE-2.0 Unless required by applicable
+ * law or agreed to in writing, software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ * for the specific language governing permissions and limitations under the License.
+ */
+
+package com.google.cloud.bigtable.grpc.scanner;
+
+import static com.google.cloud.bigtable.grpc.scanner.ReadRowTestUtils.createContentChunk;
+import static com.google.cloud.bigtable.grpc.scanner.ReadRowTestUtils.createReadRowsResponse;
+import static com.google.cloud.bigtable.grpc.scanner.ReadRowTestUtils.generateReadRowsResponses;
+import static com.google.cloud.bigtable.grpc.scanner.ReadRowTestUtils.randomBytes;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import com.google.bigtable.v1.Column;
+import com.google.bigtable.v1.Family;
+import com.google.bigtable.v1.ReadRowsResponse;
+import com.google.bigtable.v1.Row;
+import com.google.bigtable.v1.ReadRowsResponse.Chunk;
+import com.google.common.base.Function;
+import com.google.protobuf.ByteString;
+
+public class RowMergerTest {
+
+  public static Function<List<ReadRowsResponse>, List<Row>> TO_ROWS_LIST_FUNCTION =
+      new Function<List<ReadRowsResponse>, List<Row>>() {
+        @Override
+        public List<Row> apply(List<ReadRowsResponse> input) {
+          List<Row> results = new ArrayList<>();
+          Iterator<ReadRowsResponse> iterator = input.iterator();
+          while (iterator.hasNext()) {
+            results.add(RowMerger.readNextRow(iterator));
+          }
+          return results;
+        }
+      };
+
+  private static void checkKeysAreEqual(List<ReadRowsResponse> responses, List<Row> rows) {
+    Assert.assertEquals(responses.size(), rows.size());
+    for (int i = 0; i < responses.size(); i++) {
+      Assert.assertEquals(responses.get(i).getRowKey(), rows.get(i).getKey());
+    }
+  }
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Test
+  public void resultsAreReadable() throws IOException {
+    List<ReadRowsResponse> responses = generateReadRowsResponses("rowKey-%s", 3);
+    List<Row> rows = TO_ROWS_LIST_FUNCTION.apply(responses);
```

Comment:
I'll move the TO_ROWS_LIST_FUNCTION and checkKeysAreEqual here, since we don't reuse them.